plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/binary/KSPropertyGetterDescriptorImpl.kt

```diff
@@ -8,22 +8,40 @@ package org.jetbrains.kotlin.ksp.symbol.impl.binary
 import org.jetbrains.kotlin.descriptors.PropertyGetterDescriptor
 import org.jetbrains.kotlin.ksp.symbol.*
 import org.jetbrains.kotlin.ksp.symbol.impl.KSObjectCache
+import org.jetbrains.kotlin.ksp.symbol.impl.findPsi
+import org.jetbrains.kotlin.ksp.symbol.impl.kotlin.KSPropertyDeclarationImpl
+import org.jetbrains.kotlin.ksp.symbol.impl.kotlin.KSPropertyDeclarationParameterImpl
 import org.jetbrains.kotlin.ksp.symbol.impl.toFunctionKSModifiers
 import org.jetbrains.kotlin.ksp.symbol.impl.toKSModifiers
+import org.jetbrains.kotlin.psi.KtParameter
+import org.jetbrains.kotlin.psi.KtProperty
 
 class KSPropertyGetterDescriptorImpl private constructor(val descriptor: PropertyGetterDescriptor) : KSPropertyGetter {
     companion object : KSObjectCache<PropertyGetterDescriptor, KSPropertyGetterDescriptorImpl>() {
         fun getCached(descriptor: PropertyGetterDescriptor) = cache.getOrPut(descriptor) { KSPropertyGetterDescriptorImpl(descriptor) }
     }
 
-    override val origin = Origin.CLASS
+    override val origin by lazy {
+        if (descriptor.correspondingProperty.findPsi() != null) Origin.SYNTHETIC else Origin.CLASS
+    }
 
     override val location: Location = NonExistLocation
 
     override val annotations: List<KSAnnotation> by lazy {
         descriptor.annotations.map { KSAnnotationDescriptorImpl.getCached(it) }
     }
 
+    override val owner: KSPropertyDeclaration by lazy {
+        val correspondingPropertyDescriptor = descriptor.correspondingProperty
+        correspondingPropertyDescriptor.findPsi()?.let {
+            when (it) {
+                is KtProperty -> KSPropertyDeclarationImpl.getCached(it)
```

Comment:
just to be sure, local variables are modeled by KtProperty, right?

Reply:
Yes.